### PR TITLE
feat: Add deploymentIndexer fragment

### PR DESCRIPTION
**Pull request by bighammerw**
## Description

Ticket [SQN-1301](https://onfinality.atlassian.net/browse/SQN-1301?atlOrigin=eyJpIjoiZGJlMGZmMzk0MzNkNDY1ODliZWRjYjk2OWQzOGFhZTgiLCJwIjoiaiJ9)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Changes
- flex plan query
- flex plan fragment
- deployment indexers node

### CODE REVIEW

File: packages/network-query/queries/network/deployments.gql [hunk 1]
```diff
@@ -21,6 +21,27 @@ fragment DeploymentIndexerFields on DeploymentIndexer {
   }
 }
 
+fragment DeploymentIndexerNodeFields on DeploymentIndexer {
```
Comment (HZHAIX) [inline packages/network-query/queries/network/deployments.gql:24]: Reuse the `IndexerDeploymentFields` instead duplicate the fields, recommend to have a look on https://www.apollographql.com/docs/react/data/fragments/

```
fragment DeploymentIndexerNodeFields on DeploymentIndexer {
  ...DeploymentIndexerFields
  deployment {
    id
    project {
      id
      metadata
    }
  }
}
```

File: packages/network-query/queries/network/flexplan.gql [hunk 2]
```diff
@@ -0,0 +1,64 @@
+# Copyright 2020-2022 SubQuery Pte Ltd authors & contributors
+# SPDX-License-Identifier: Apache-2.0
+
+fragment StateChannelFields on StateChannel {
```
Comment (HZHAIX) [inline packages/network-query/queries/network/flexplan.gql:4]: This is duplicate with `stateChannel.gql`

File: packages/network-query/queries/network/deployments.gql (hunk 1 above)
Comment (lamcc21) [reply packages/network-query/queries/network/deployments.gql:24]: Don't include any fragments unless they are being used in a query inside network-query package

File: packages/network-query/queries/network/statechannel.gql [hunk 3]
```diff
@@ -22,6 +22,15 @@ fragment StateChannelFields on StateChannel {
...
+fragment StateChannelsFields on StateChannel {
+  stateChannels {
+    totalCount
+    nodes {
+      ...StateChannelFields
+    }
+  }
+}
+
 query GetPaygPlans($indexer: String!, $deploymentId: String!, $status: ChannelStatus!) {
```
Comment (lamcc21) [inline packages/network-query/queries/network/statechannel.gql:34]: this fragment is not being used anywhere in this file

**Review by CaiYiLiang — APPROVED** [review]

**Pull request merged**